app/src/main/java/org/fossasia/openevent/app/module/organizer/password/ChangePasswordFragment.java

```diff
@@ -0,0 +1,121 @@
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import org.fossasia.openevent.app.OrgaApplication;
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.common.app.lifecycle.view.BaseFragment;
+import org.fossasia.openevent.app.common.utils.ui.ViewUtils;
+import org.fossasia.openevent.app.databinding.ChangePasswordFragmentBinding;
+import org.fossasia.openevent.app.module.organizer.password.contract.IChangePasswordPresenter;
+import org.fossasia.openevent.app.module.organizer.password.contract.IChangePasswordView;
+
+import javax.inject.Inject;
+
+import br.com.ilhasoft.support.validation.Validator;
+import dagger.Lazy;
+
+import static org.fossasia.openevent.app.common.utils.ui.ViewUtils.showView;
+
+public class ChangePasswordFragment extends BaseFragment<IChangePasswordPresenter> implements IChangePasswordView {
+
+    @Inject
+    Lazy<IChangePasswordPresenter> presenterProvider;
+
+    private ChangePasswordFragmentBinding binding;
+    private Validator validator;
+
+    public static ChangePasswordFragment newInstance() {
+        return new ChangePasswordFragment();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        OrgaApplication
+            .getAppComponent()
+            .inject(this);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = DataBindingUtil.inflate(inflater, R.layout.change_password_fragment, container, false);
+        validator = new Validator(binding);
+
+        AppCompatActivity activity = ((AppCompatActivity) getActivity());
+        activity.setSupportActionBar(binding.toolbar);
+
+        ActionBar actionBar = activity.getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setHomeButtonEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        return binding.getRoot();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        getPresenter().attach(this);
+        binding.setOrganizerPassword(getPresenter().getChangePasswordObject());
+        getPresenter().start();
+
+        binding.btnChangePassword.setOnClickListener(view -> {
+            if (!validator.validate())
+                return;
+
+            if (!binding.newPassword.getText().toString()
+                .equals(binding.confirmNewPassword.getText().toString())) {
+
+                showError("Passwords Do not Match");
+                return;
+            }
+
+            String url = binding.url.baseUrl.getText().toString().trim();
+            getPresenter().setBaseUrl(url, binding.url.overrideUrl.isChecked());
+            getPresenter().changePasswordRequest();
```

Comment:
Send the password to the presenter as text, we want testability. So we want a method which we can use to test the behaviour once a string is passed.

Reply:
I didn't get it. We can make a test for ChangePasswordPresenter in which we can mock the ChangePassword object and then test the changePassword () method. What do you wanna say? Do you mean checking the current password as a valid one or not by passing it in presenter?

Reply:
We can mock it but how will we inject it. And also how will we do it for many cases of failures and success?